improver_tests/acceptance/test_expected_value.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown copyright. The Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Tests for the expected-value CLI."""
+
+
+import iris
+import numpy as np
+import pytest
+
+from . import acceptance as acc
+
+pytestmark = [pytest.mark.acc, acc.skip_if_kgo_missing]
+CLI = acc.cli_name_with_dashes(__file__)
+run_cli = acc.run_cli(CLI)
+
+
+@pytest.mark.parametrize(
+    "input_kind,atol,significant_diff_fraction",
+    [("realization", 0.001, 0.0), ("percentile", 0.09, 0.0), ("threshold", 0.18, 0.2)],
```

Comment:
The KGO is calculated directly from the realization data. I checked the KGO by doing a similar mean calculation with `ncwa` from NetCDF Operators and the data grid was identical (via `ncdiff`).

The percentile data input for the acceptance tests is derived from the realization data via something like
```
improver generate-percentiles --coordinates realization --percentiles 5,10,15,20,25,30,35,40,45,50,55,60,65,70,75,80,85,90,95 realization.nc --output percentile.nc
```

There are 19 percentiles derived from 18 realizations. The realization -> percentile conversion is like a resampling/regridding, but in the probability representation dimension rather than spatial dimensions. There's some associated loss of precision in the percentile input to this CLI, so when comparing against the KGO (derived from the realization representation) there are differences resulting in needing a larger `atol`.

The `atol` for realization could be reduced to zero (ie. identical) as the operation is simple and well defined in floating point maths. I chose 0.001 kelvin as a not-scientifically-significant for temperature data tolerance.

Reply:
Thanks. I figured that was the case, but thought it would be worthwhile confirming this was the case.